### PR TITLE
CB-18829 Setup FreeIPA LDAP agent

### PR DESCRIPTION
**Pull request by lacikaaa**
See detailed description in the commit message.

### CODE REVIEW

**Review by handavid — APPROVED** [review]
this all looks good to me.

File: freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/config/LdapAgentConfigProvider.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+@Component
+public class LdapAgentConfigProvider {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LdapAgentConfigProvider.class);
+
+    private static final String LDAP_AGENT_CONFIG_KEY = "ldapagent";
+
+    private static final String LDAP_AGENT_PILLAR_PATH = "/ldapagent/init.sls";
+
+    public Map<String, SaltPillarProperties> generateConfig(String domain) {
```
[the remark this answers is not in the document]
Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/config/LdapAgentConfigProvider.java:23]: I can, but this should never ever be null

[the remark this answers is not in the document]
Comment (lacikaaa) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/config/LdapAgentConfigProvider.java:23]: what do you thin about adding `Objects.requireNonNull(domain);` in the first line?

[the remark this answers is not in the document]
Comment (tiborpopovics) [reply freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/config/LdapAgentConfigProvider.java:23]: It's fine for me, thanks!

**Review by tiborpopovics — APPROVED** [review]

**Pull request merged**